packages/ui-components/src/components/SplitButton.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Grid from '@material-ui/core/Grid';
+import MuiButtonGroup from '@material-ui/core/ButtonGroup';
+import KeyboardArrowDown from '@material-ui/icons/KeyboardArrowDown';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import Grow from '@material-ui/core/Grow';
+import Paper from '@material-ui/core/Paper';
+import Popper from '@material-ui/core/Popper';
+import MenuItem from '@material-ui/core/MenuItem';
+import MenuList from '@material-ui/core/MenuList';
+import { Button } from './Button';
+
+const DefaultButton = styled(Button)`
+  padding-left: 0.5rem;
+  padding-right: 0.5rem;
+
+  &.MuiButtonBase-root {
+    margin: 0;
+  }
+`;
+
+const ButtonGroup = styled(MuiButtonGroup)`
+  margin-right: 1rem;
+`;
+
+export const SplitButton = ({ options, selectedId, setSelectedId, onClick, ButtonComponent }) => {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef(null);
+
+  const selectedOption = options.find(o => o.id === selectedId);
+
+  const handleClick = () => {
+    onClick(selectedId);
+  };
+
+  const handleMenuItemClick = (event, id) => {
+    setSelectedId(id);
+    setOpen(false);
+  };
+
+  const handleToggle = () => {
+    setOpen(prevOpen => !prevOpen);
+  };
+
+  const handleClose = event => {
+    if (anchorRef.current && anchorRef.current.contains(event.target)) {
```

Comment:
What's happening here? Could you add a comment for front end noobs like me?

Reply:
Oh worked it out
```suggestion
    // if the user clicks away, but their click still lands on some part of the component, keep open
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
```

Reply:
Copied comment to my commit